tests/contrib/rediscluster/test.py

```diff
@@ -139,3 +138,25 @@ def test_user_specified_service(self):
         assert len(spans) == 1
         span = spans[0]
         assert span.service != "mysvc"
+
+    @TracerTestCase.run_in_subprocess(env_overrides=dict(DD_REDISCLUSTER_SERVICE="myrediscluster"))
+    def test_env_user_specified_rediscluster_service(self):
+        self.r.get("cheese")
+        span = self.get_spans()[0]
+        assert span.service == "myrediscluster", span.service
+
+    @TracerTestCase.run_in_subprocess(
+        env_overrides=dict(DD_SERVICE="app-svc", DD_REDISCLUSTER_SERVICE="myrediscluster")
+    )
+    def test_service_precedence(self):
+        self.r.get("cheese")
+        span = self.get_spans()[0]
+        assert span.service == "myrediscluster"
+
+        self.reset()
+
+        # Do a manual override
+        Pin.override(self.r, service="myrediscluster", tracer=self.tracer)
+        self.r.get("cheese")
+        span = self.get_spans()[0]
+        assert span.service == "myrediscluster", span.service
```

Comment:
Just curious, what are we testing here exactly?

Reply:
yep, unless I missed something, that portion is supposed to make sure that we use the ```DD_REDISCLUSTER_SERVICE``` env variable instead of the ```DD_SERVICE``` when both are declared.

Reply:
I see, is there any specific reason we have to manually override the Pin again after we reset the tracer?

Reply:
I'm just not sure what lines 159-162 are testing that are different from the previous test

Reply:
You're right 👍 I don't think there would be a situation here where the dd_service is defined and the ```DD_REDISCLUSTER_SERVICE``` env variable is not taking over in precedence